panels/models.py

```diff
@@ -10,6 +18,42 @@ def panel_applicants(self):
         return self.query(PanelApplicant).options(joinedload(PanelApplicant.application)).order_by('first_name', 'last_name')
 
 
+class SocialMediaMixin(JSONColumnMixin('social_media', c.SOCIAL_MEDIA)):
+    social_media_urls = config['social_media_urls']
+    social_media_placeholders = config['social_media_placeholders']
+
+    @classmethod
+    def get_placeholder(cls, name):
+        name = cls.unqualify(name)
+        return cls.social_media_placeholders.get(name, '')
+
+    @property
+    def has_social_media(self):
+        return any(getattr(self, f) for f in self.social_media_fields.keys())
+
+    def __getattr__(self, name):
+        if name.endswith('_url'):
+            field_name = self.unqualify(name[:-4])
+            if field_name in self.social_media_fields:
+                attr = super(SocialMediaMixin, self).__getattr__(field_name)
+                attr = attr.strip('@#?=. ') if attr else ''
```

Comment:
Ok, actually, that was correct as it was. If `attr` is `None` then we can't call `strip()` on it. And if `attr` consists entirely of these characters `@#?=. ` (including spaces), then we want to return an empty string.

So we do the first if statement to handle the case where `attr` is `None`, and the second if statement to handle the case where `attr` is stripped down to the empty string.